assume/units/powerplant.py

```diff
@@ -144,7 +144,11 @@ def init_marginal_cost(self):
         """
         Initialize the marginal cost of the unit.
         """
-        self.marginal_cost = self.calc_simple_marginal_cost()
+
+        if not self.partial_load_eff:
+            self.marginal_cost = self.calc_simple_marginal_cost()
+        else:
+            self.marginal_cost = None
```

Comment:
It think it would look better to not negate the logic?

Why should None be returned if partial_load_eff is set?

Reply:
because if it is set and not None, cal_marginal_cost function returns this value and ignores the partial efficiency.

Reply:
or we should fix the calc_marginal_cost function, either one is valid I believe

Reply:
@maurerle so what do you think?